### ISSUE TITLE
Adding Carla support to appimage build of LMMS

### ISSUE THREAD

**Issue by vlad1777d** [LMMS/lmms#3976]
Hello dear friends.

LMMS 1.1.3 worked well with Carla, but new RC4 appimage build doesn't see it.
Maybe is there a way to add support for Carla to LMMS 1.2.0 ?
I never compiled LMMS itself, so I can't tell details.
Please, @tresf , correct me if I told something wrong.
@falkTX could you tell about: how difficult it can be ?

Thanks.
Best regards,
Vladislav

**Comment by falkTX** [comment]
I don't really know what is involved, since I never built Carla as appimage myself.
Carla's use of PyQt for the frontend might make things complicated, not sure...

**Comment by tresf** [comment]
> Carla's use of PyQt for the frontend might make things complicated, not sure...

It does, but it's also rather large in size, so for now I don't think it's a good candidate for LMMS to start bundling so we'll just assume it's installed.

I have a PR ready (#4026) where Carla loads, but I haven't done any actually functionality testing yet.  Testing appreciated.

![image](https://user-images.githubusercontent.com/6345473/33418680-07c3e390-d575-11e7-8994-ad5b29cb94d4.png)

**Comment by tresf** [comment]
@vlad1777d can you please test this latest build?  https://github.com/tresf/lmms/releases/download/v1.2.0-rc4/lmms-1.2.0-rc4.112-linux-x86_64.AppImage

**Comment by vlad1777d** [comment]
@tresf, of course, I will.
Maybe not today, but soon I'll continue making track on it.

Are there any incompatibilities between LMMS 1.1.3 file format and 1.2.0 ?

Because at first I created my project in 1.2.0, than when I failed to load window with ladspa plugins, I continued it in 1.1.3, than just ran on 1.2.0 again.
Can I loss some data so ?

**Comment by tresf** [comment]
There shouldn't be any major compat issues between 1.1 and 1.2.  if you find some please file a bug report worth a sample project so we can debug.

**Comment by falkTX** [comment]
lmms-1.2.0-rc4.112-linux-x86_64.AppImage works for me here, but the GUI is scaled 1.5 to 2x which makes it unusable...
when I move the window to my other screen, everything gets too tiny instead (1.5x smaller than my usual desktop).
this is quite strange, as I have no scaling whatsoever enabled on my system (KDE5 neon).

**Comment by lukas-w** [comment]
> the GUI is scaled 1.5 to 2x which makes it unusable...

@tresf We may want to revisit 8bc9e9d9b4785c47a2c809bdc7de907f11381c04…

Most likely unrelated, but that AppImage segfaults for me on start with my `.lmmsrc.xml`:

```
<?xml version="1.0"?>
<!DOCTYPE lmms-config-file>
<lmms version="1.2.0-rc4">
  <MidiAlsaRaw device="default"/>
  <MidiJack device="lmms"/>
  <Midialsaseq device="default"/>
  <app language="en" configured="1" nommpz="0" nomsgaftersetup="0" disablebackup="0" displaydbfs="0" openlastproject="0"/>
  <audioalsa device="default" channels="2"/>
  <audiojack clientname="lmms" channels="2"/>
  <audiooss device="/dev/dsp" channels="2"/>
  <audiopa device="default" channels="2"/>
  <audioportaudio backend="ALSA" device="HDA Intel PCH: ALC892 Analog (hw:0,0)"/>
  <audiosdl device=""/>
  <midioss device="/dev/midi"/>
  <mixer hqaudio="0" framesperaudiobuffer="256" mididev="ALSA-Sequencer (Advanced Linux Sound Architecture)" audiodev="PortAudio"/>
  <paths stkdir="/tmp/.mount_lmms-1IHnj3c/usr/bin/../share/stk/rawwaves/" workingdir="/home/lukas/lmms/" vstdir="/home/lukas/lmms/plugins/vst/" artwork="data:/themes/default/" backgroundartwork="" gigdir="/home/lukas/lmms/samples/gig/" laddir="/home/lukas/lmms/plugins/ladspa/" sf2dir="/home/lukas/lmms/samples/sf2/" defaultsf2=""/>
  <tooltips disabled="0"/>
  <ui saveinterval="2" displaywaveform="0" syncvstplugins="0" enablerunningautosave="1" enableautosave="1" animateafp="1" printnotelabels="0" compacttrackbuttons="0" disableautoquit="0" oneinstrumenttrackwindow="0" smoothscroll="0"/>
  <recentfiles>
    <file path="data:/projects/templates/default.mpt"/>
  </recentfiles>
</lmms>
```

```
#0  0x0000000000594eb3 in SetupDialog::SetupDialog(SetupDialog::ConfigTabs) ()
#1  0x0000000000588f0a in MainWindow::finalize() ()
#2  0x000000000057b8f8 in GuiApplication::GuiApplication() ()
#3  0x000000000049a730 in main ()
```

Removing `.lmmsrc.xml` fixes it.

**Comment by tresf** [comment]
> @tresf We may want to revisit 8bc9e9d…

Yeah, I've disabled it in my PR.  It's clearly broken in Qt, it can be toggled back on manually.  I've added the instructions in the Carla PR.

>  that AppImage segfaults for me on start with my .lmmsrc.xml:

I think we have a bug with our sound device selection.  I've had it several times with 1.2.0 but haven't isolated the cause.  If someone else can shed light on what causes this, we can open a dedicated bug report for it.
